portfolio/src/main/webapp/script.js

```diff
@@ -12,16 +12,64 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random rugby player's name to the page.
- */
-function addRandomPlayer() {
-  const players =
-      ['Beast', 'Siya Kolisi', 'Latica Nela', 'Chipo Mupeso', 'Gareth Schreuder', 'Andrew James', 'Niaan'];
-  // Pick a random player.
-  const player = players[Math.floor(Math.random() * players.length)];
-
-  // Add it to the page.
-  const playerContainer = document.getElementById('greeting-container');
-  playerContainer.innerText = player;
-}
\ No newline at end of file
+/** Fetches comments from the server and adds them to the DOM. */
+function loadCommentSection() {
+  fetch('/data').then(response => response.json()).then(data => {
+    const commentElements = document.getElementById('comments');
+    commentElements.innerText = '';
+    const limit = document.getElementById('commentLimit').value;
+    const limitMap = new Map();
+    var count = 0;
+
+    data.forEach(comment => {
+      if (count < limit){
+        commentElements.appendChild(createCommentElement(comment));
+        count++;
+      } 
+      var mapKey = comment.commentLimit;
+      var currentLimit = limitMap.has(mapKey) ? limitMap.get(mapKey) : 0;
```

Comment:
I'm not sure a map is the right choice here. Let's discuss this offline tomorrow, if you have time. I still approved the PR though, don't want you to be blocked on this.

Reply:
Okay sure, we'll chat today.